x-pack/plugins/data_enhanced/public/search/sessions_mgmt/types.ts

```diff
@@ -4,11 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { SearchSessionStatus } from '../../../common';
+import { SearchSessionSavedObjectAttributes, SearchSessionStatus } from '../../../common';
 import { ACTION } from './components/actions';
 
 export const DATE_STRING_FORMAT = 'D MMM, YYYY, HH:mm:ss';
 
+/**
+ * Some properties are optional for a non-persisted Search Session.
+ * This interface makes them mandatory, because management only shows persisted search sessions.
+ */
+export interface PersistedSearchSessionSavedObjectAttributes
+  extends SearchSessionSavedObjectAttributes {
```

Comment:
nit:

I would probably prefer something like instead: 

```
type PersistedSearchSessionSavedObjectAttributes = SearchSessionSavedObjectAttributes & Required<Pick< SearchSessionSavedObjectAttributes , 'name' | 'appId' | 'urlGeneratorId ' | 'initialState' | 'restoreState'>>
```

With the current interface extend approach, if we need to do change to those places,  it is easier to make a mistake by updating the type in one place and forgetting the other as typescript won't budge in case of mismatch.



Also less duplication